framework/ChiPhysics/PhysicsMaterial/transportxsections/prop_transportxs_01b_finalize.cc

```diff
@@ -68,131 +80,192 @@ void chi_physics::TransportCrossSections::Finalize()
-                "or positive. Zero values must be accounted allowed "
-                "for photo-fission, which is a threshold reaction. Prompt "
-                "fission neutron yields must be greater than 1 or 0.");
-
-          //compute other quantities
-          for (unsigned int g = 0; g < num_groups; ++g)
-          {
-            nu[g] = nu_prompt[g] + nu_delayed[g];
-            beta[g] = nu_delayed[g] / nu[g];
-          }
-        }//if nu_prompt and nu_delayed specified
-        else if (was_specified(nu) &&
-                 was_specified(beta))
-        {
-          if (!std::all_of(nu.begin(), nu.end(),
-                           [](double x) { return x == 0.0 || x > 1.0; }))
-            throw std::logic_error(
-                "Fission neutron yield must be greater than 1 or 0.");
-          if (!std::all_of(beta.begin(), beta.end(),
-                           [](double x) { return x >= 0.0 && x <= 1.0; }))
-            throw std::logic_error(
-                "Delayed neutron fractions must be in the range [0.0, 1.0].");
-
-          //compute other quantities
-          for (unsigned int g = 0; g < num_groups; ++g)
-          {
-            nu_prompt[g] = (1.0 - beta[g]) * nu[g];
-            nu_delayed[g] = beta[g] * nu[g];
-          }
-
-          if (was_specified(nu_sigma_f))
-            for (unsigned int g = 0; g < num_groups; ++g)
-              if (nu[g] != 0.0)
-                sigma_f[g] = nu_sigma_f[g] / nu[g];
-        } //if nu and beta specified
-        else
+        if (!std::all_of(nu_prompt.begin(), nu_prompt.end(),
+                         [](double x) { return x == 0.0 || x > 1.0; }) &&
+            !std::all_of(nu_delayed.begin(), nu_delayed.end(),
+                         [](double x) { return x >= 0.0; }))
           throw std::logic_error(
-              "Invalid specification of fission neutron yield data.");
+              "Invalid prompt and delayed fission neutron yields "
+              "encountered.\nPrompt fission neutron yields must be either "
+              "zero or greater than one.\nDelayed fission neutron yields "
+              "must be zero or greater.");
 
         //compute other quantities
+        nu.assign(num_groups, 0.0);
+        beta.assign(num_groups, 0.0);
         for (unsigned int g = 0; g < num_groups; ++g)
         {
-          nu_sigma_f[g] = nu[g] * sigma_f[g];
-          nu_prompt_sigma_f[g] = nu_prompt[g] * sigma_f[g];
-          nu_delayed_sigma_f[g] = nu_delayed[g] * sigma_f[g];
+          nu[g] = nu_prompt[g] + nu_delayed[g];
+          beta[g] = nu_delayed[g] / nu[g];
         }
-      }
+      }//if prompt/delayed specified
 
-      //check and normalize prompt fission spectrum
+      else if (is_valid(nu) && is_valid(beta))
       {
-        //throw error if not specified
-        if (!was_specified(chi_prompt))
+        if (!std::all_of(nu.begin(), nu.end(),
+                         [](double x) { return x == 0.0 || x > 1.0; }))
```

Comment:
No need. If we did it would be a lambda in this routine and any benefit we may gain from that doesn't matter because this is run during initialization.